src/profile-logic/marker-schema.js

```diff
@@ -0,0 +1,217 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+// @flow
+import type { MarkerFormatType } from 'firefox-profiler/types';
+import {
+  formatNumber,
+  formatPercent,
+  formatBytes,
+  formatSeconds,
+  formatMilliseconds,
+  formatTimestamp,
+  formatMicroseconds,
+  formatNanoseconds,
+} from '../utils/format-numbers';
+
+/**
+ * TODO - These will eventually be stored in the profile, but for now
+ * define them here.
+ */
+export const markerSchema = [
+  {
+    name: 'Bailout',
+    display: ['marker-chart', 'marker-table'],
+    data: [
+      { key: 'bailoutType', label: 'Type', format: 'string' },
+      { key: 'where', label: 'Where', format: 'string' },
+      { key: 'script', label: 'Script', format: 'string' },
+      { key: 'functionLine', label: 'Function Line', format: 'integer' },
+      { key: 'bailoutLine', label: 'Bailout Line', format: 'integer' },
+    ],
+  },
+  {
+    name: 'GCMajor',
+    display: ['marker-chart', 'marker-table', 'timeline-memory'],
+    data: [
+      // Use custom handling
```

Comment:
I wish there could be an indicator other than an empty array for custom handlings. Not sure what would it be though.

Reply:
Well, you can do both custom handling AND use the schema, so they aren't mutually exclusive.

Reply:
Ah, yeah that's right. I wrote it actually before reviewing all the changes. It's good to keep it that way then :)